auslib/web/views/client.py

```diff
@@ -70,40 +73,55 @@ def get(self, **url):
+                    response_release = dbo.releases.getReleaseBlob(name=blob_name)
+                    if not response_release:
+                        self.log.warning("No release found with name: %s", blob_name)
+                        continue
+
+                    response_blobs.append({'product_query': product_query,
+                                           'response_release': response_release,
+                                           'response_update_type': update_type})
             else:
                 response_blobs.append({'product_query': query,
                                        'response_release': release,
                                        'response_update_type': update_type})
 
-            xml = ['<?xml version="1.0"?>']
-            xml.append('<updates>')
-
-            # We only sample the first blob for the header and footer, since we
-            # assume that all blobs will have similar ones. We might want to
+            # getHeaderXML() returns outermost header for an update which
+            # is same for all release type
+            xml = release.getHeaderXML()
+            # we assume that all blobs will have similar ones. We might want to
             # verify that all of them are indeed the same in the future.
 
-            # Sampling the footer from the first blob
-            headerXML = response_blobs[0]['response_release'].getHeaderXML(response_blobs[0]['product_query'],
-                                                                           response_blobs[0]['response_update_type'],
-                                                                           app.config["WHITELISTED_DOMAINS"],
-                                                                           app.config["SPECIAL_FORCE_HOSTS"])
-            # Sampling the footer from the first blob
-            footerXML = response_blobs[0]['response_release'].getFooterXML(response_blobs[0]['product_query'],
-                                                                           response_blobs[0]['response_update_type'],
-                                                                           app.config["WHITELISTED_DOMAINS"],
-                                                                           app.config["SPECIAL_FORCE_HOSTS"])
-            if headerXML:
-                xml.append(headerXML)
+            # Extracting Header
+            # In case of superblob Extracting Header form parent release
+            innerHeaderXML = release.getInnerHeaderXML(query,
+                                                       update_type,
+                                                       app.config["WHITELISTED_DOMAINS"],
+                                                       app.config["SPECIAL_FORCE_HOSTS"])
+            # Extracting Footer
+            # In case of superblob Extracting Header form parent release
+            innerFooterXML = release.getInnerFooterXML(query,
```

Comment:
@mozbhearsum 
Used intermediate variable to check if None is returned ( None is returned in case of systemaddons blob having 'uninstall' as False https://github.com/mozilla/balrog/blob/master/auslib/blobs/systemaddons.py#L83 )